app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
```

Comment:
If I set it to the structure you suggest, I will need to check if search is disabled every single time, whether or not getSeveralRandomIds returns a null. (Since search is almost never disabled, I think this will be worse for performance)